lms/djangoapps/courseware/tests/test_views.py

```diff
@@ -2565,7 +2595,10 @@ def test_should_show_enroll_button(self, course_open_for_self_enrollment,
             patch_is_masters_only.return_value = is_masters_only
             course.invitation_only = invitation_only
 
-            self.assertEqual(views.CourseTabView.should_show_enroll_button(course), expected_should_show_enroll_button)
+            self.assertEqual(
+                views.CourseTabView.course_open_for_learner_enrollment(course),
+                expected_should_show_enroll_button
```

Comment:
Nit: Would you mind renaming the functions and statements that contains the text `should_show_enroll_button` to use `course_open_for_learner_enrollment` in order to match the change to the function name? For example, in this function, the name `test_should_show_enroll_button` -> `test_ course_open_for_learner_enrollment` and `expected_should_show_enroll_button` -> `expected_course_open_for_learner_enrollment`.

Reply:
I'm not actually sure that the mismatch makes things more unclear; the fact that the initial intent of the choice of name was to refer to another identifier doesn't mean that it's now referring to the wrong thing. Are any semantic links broken, from what you can see?

Reply:
This test is validating that the `course_open_for_learner_enrollment` method returns the correct value, so I feel that the name of the test should reflect what it is testing. In this case, if we're just testing that one function, then I think that an agreement between the test name and the thing being tested is appropriate. I agree that that is not always the case, but I do feel like it's warranted here.

When looking at this test, I find it confusing that the name is `test_should_show_enroll_button`, because I don't see a reference to anything concerning showing an enrollment button (aside from the function parameter name). I know that `test_should_show_enroll_button` is referring to testing the  `course_open_for_learner_enrollment` method, because I know that `should_show_enroll_button` was changed to `course_open_for_learner_enrollment`, but I don't think people looking at this in the future will make that connection because they won't know of this change.

Reply:
How do you think they would read it? My inclination is that folks would glean pretty quickly from the structure of the test that it's testing the return value of this particular function. Then from the variable names used/the test name they could glean further that the function being tested is used in a particular way. This test, in my view, expresses more to the reader as written than as proposed. Does that make sense? Happy to chat offline (or to agree to disagree! Either way is fine).